plugins/inputs/rethinkdb/rethinkdb_server.go

```diff
@@ -174,23 +174,27 @@ func (s *Server) addTableStats(acc telegraf.Accumulator) error {
 		return errors.New("could not parse table_status results")
 	}
 	for _, table := range tables {
-		cursor, err := gorethink.DB("rethinkdb").Table("stats").
-			Get([]string{"table_server", table.ID, s.serverStatus.ID}).
-			Run(s.session)
-		if err != nil {
-			return fmt.Errorf("table stats query error, %s", err.Error())
-		}
-		defer cursor.Close()
-		var ts tableStats
-		if err := cursor.One(&ts); err != nil {
-			return fmt.Errorf("failure to parse table stats, %s", err.Error())
-		}
+		return func() error {
```

Comment:
While being a matter of style, I don't like those function constructs to allow `defer` in the loop as it make the code look complex. Can we please move the inner code to an own (named) function instead?

Reply:
Moved.